server/controllers/DestinationController.js

```diff
@@ -0,0 +1,227 @@
+/* eslint-disable node/no-unsupported-features/es-syntax */
+const DestinationModel = require('../models/DestinationModel')
+const getLatLon = require('../googleapi/getLatLon')
+const generateDestination = require('../openai/generateDestination')
+const nlp = require('compromise')
+
+class DestinationController {
+  async findClosestDestinations(latitude, longitude, n) {
+    try {
+      const milesToMeters = 1609.34 // Conversion factor
+      const maxDistance = 30 * milesToMeters // Limit search to 30 miles
+
+      const closestDestinations = await DestinationModel.find({
```

Comment:
I was gonna say controllers shouldnt have access to the models but then I remembered our controller is that layer lol